### PR TITLE
Fix macOS builds

### LINKED ISSUE #48: Fix macOS builds
Need to sanity check that they look fine and can be run.

Windows and Linux builds have been tested, and the web build can be tested by anyone on itch.io.

### PR DESCRIPTION
**Pull request by janhohenheim**
Resolves #48

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Once this is validated please backport it to the CI template.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -233,26 +235,19 @@ jobs:
           tag: ${{ github.ref }}
           overwrite: true
 
-  # Build for MacOS x86_64.
-  build-for-macOS-intel:
-    runs-on: macOS-latest
+  # Build for MacOS x86_64 and Apple Silicon
```
Comment (benfrankel) [inline .github/workflows/release.yaml:238]:
```suggestion
  # Build for MacOS x86_64 and Apple Silicon.
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -233,26 +235,19 @@ jobs:
...
+        VERSION: ${{ needs.get-version.outputs.version }}
     steps:
       - name: Set PACKAGE environment variable
         run: |
-          echo "PACKAGE=${{ env.PACKAGE_NAME }}-${{ env.VERSION }}-${{ env.PLATFORM }}" >> "${GITHUB_ENV}"
-
-      - name: Set OUT_DIR environment variable
-        run: |
-          echo "OUT_DIR=build/${{ env.PACKAGE }}/${{ env.PACKAGE_NAME }}.app/Contents/MacOS" >> "${GITHUB_ENV}"
+          "PACKAGE=${{ env.PACKAGE_NAME }}-${{ env.VERSION }}-${{ env.PLATFORM }}" | Out-File -FilePath "${Env:GITHUB_ENV}" -Append
```
Comment (benfrankel) [inline .github/workflows/release.yaml:250]: I'm curious. This is powershell syntax, but `linux` and `macos-latest` runners are said to use `bash`.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -262,103 +257,67 @@ jobs:
...
+      - name: Upload release
         uses: svenstaro/upload-release-action@v2
         with:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: build/${{ env.PACKAGE }}.dmg
+          file: ${{ env.PACKAGE }}.dmg
           asset_name: ${{ env.PACKAGE }}.dmg
           release_name: ${{ env.VERSION }}
-          tag: ${{ github.ref }}
+          tag: ${{ env.VERSION }}
```
Comment (benfrankel) [inline .github/workflows/release.yaml:320]: Actually I think this `github.ref` is why manually triggering the release created a `main` tag. Maybe change the other `tag:` values to this as well.

**Review by benfrankel — CHANGES_REQUESTED** [review]
Not a full review pass.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (janhohenheim) [reply .github/workflows/release.yaml:250]: 🤷‍♂️ I don't know about shells, I just followed https://github.com/rparrett/pixie_wrangler/blob/main/.github/workflows/release.yaml#L175

File: .github/workflows/release.yaml (hunk 3 above)
Comment (janhohenheim) [reply .github/workflows/release.yaml:320]: Good idea. [`1fb119d` (#112)](https://github.com/TheBevyFlock/bevy-template/pull/112/commits/1fb119d49c27e753daa7a2eba681e85639c318f6)

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -23,6 +23,8 @@ env:
   # The itch.io page to upload to, in the format: `user-name/project-name`.
   # Comment this out to disable.
   ITCH_TARGET: the-bevy-flock/bevy-template
+  # The organization or author that owns the rights to the game
```
Comment (benfrankel) [inline .github/workflows/release.yaml:26]:
```suggestion
  # The organization or author that owns the rights to the game.
```

File: .github/workflows/release.yaml (hunk 2 above)
Comment (janhohenheim) [reply .github/workflows/release.yaml]: [`c4aa286` (#112)](https://github.com/TheBevyFlock/bevy-template/pull/112/commits/c4aa286b027bd2bd8b770ba3b9f955b9f0933eed)

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -262,103 +257,67 @@ jobs:
...
-      - name: Prepare package
-        run: |
-          mkdir -p "${{ env.OUT_DIR }}"
-          cp "target/${{ env.TARGET }}/${{ env.PROFILE }}/${{ env.BINARY }}" "${{ env.OUT_DIR }}"
-          cp -r assets "${{ env.OUT_DIR }}" || true # Ignore error if `assets` folder does not exist
-
-      - name: Compress package
-        working-directory: ./build/${{ env.PACKAGE }}
+            targets: aarch64-apple-darwin
+      - name: Build release for Apple Silicon
```
Comment (benfrankel) [inline .github/workflows/release.yaml:261]:
```suggestion
            targets: aarch64-apple-darwin

      - name: Build binary for Apple Silicon
```

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -262,103 +257,67 @@ jobs:
...
-        with:
-          path: build/${{ env.PACKAGE }}.dmg
-          name: ${{ env.PLATFORM }}
-          retention-days: 1
-
-      - name: Upload package to Github release
-        if: ${{ env.UPLOAD_PACKAGES_TO_GITHUB_RELEASE == 'true' }}
-        uses: svenstaro/upload-release-action@v2
+          SDKROOT=$(xcrun -sdk macosx --show-sdk-path) cargo build --profile="${{ env.PROFILE }}" --no-default-features --target=aarch64-apple-darwin
+      - name: Install rust toolchain for Apple x86
```
Comment (benfrankel) [inline .github/workflows/release.yaml:264]:
```suggestion
          SDKROOT=$(xcrun -sdk macosx --show-sdk-path) cargo build --profile="${{ env.PROFILE }}" --no-default-features --target=aarch64-apple-darwin

      - name: Install rust toolchain for Apple x86_64
```

File: .github/workflows/release.yaml [hunk 7]
```diff
@@ -262,103 +257,67 @@ jobs:
...
-
-      - name: Install Rust toolchain
-        uses: dtolnay/rust-toolchain@stable
-        with:
-          targets: ${{ env.TARGET }}
-
-      - name: Build binary
+          toolchain: stable
+          targets: x86_64-apple-darwin
+      - name: Build release for x86 Apple
```
Comment (benfrankel) [inline .github/workflows/release.yaml:269]:
```suggestion
          targets: x86_64-apple-darwin

      - name: Build binary for Apple x86_64
```

File: .github/workflows/release.yaml [hunk 8]
```diff
@@ -262,103 +257,67 @@ jobs:
...
-          path: build/${{ env.PACKAGE }}.dmg
-          name: ${{ env.PLATFORM }}
-          retention-days: 1
-
-      - name: Upload package to Github release
-        if: ${{ env.UPLOAD_PACKAGES_TO_GITHUB_RELEASE == 'true' }}
-        uses: svenstaro/upload-release-action@v2
+          SDKROOT=$(xcrun -sdk macosx --show-sdk-path) cargo build --profile="${{ env.PROFILE }}" --no-default-features --target=aarch64-apple-darwin
+      - name: Install rust toolchain for Apple x86
+        uses: dtolnay/rust-toolchain@master
```
Comment (benfrankel) [inline .github/workflows/release.yaml:265]:
```suggestion
        uses: dtolnay/rust-toolchain@stable
```
Unless there's a reason for this?

**Review by benfrankel — CHANGES_REQUESTED** [review]
Mostly nits, will review the big blob next.

File: .github/workflows/release.yaml [hunk 9]
```diff
@@ -262,103 +257,67 @@ jobs:
...
-      - name: Build binary
+          toolchain: stable
+          targets: x86_64-apple-darwin
+      - name: Build release for x86 Apple
         run: |
-          cargo build --profile="${{ env.PROFILE }}" --target="${{ env.TARGET }}" --no-default-features
-
-      - name: Prepare package
+          SDKROOT=$(xcrun -sdk macosx --show-sdk-path) cargo build --profile="${{ env.PROFILE }}" --no-default-features --target=x86_64-apple-darwin
+      - name: Create Universal Binary
```
Comment (benfrankel) [inline .github/workflows/release.yaml:272]:
```suggestion
          SDKROOT=$(xcrun -sdk macosx --show-sdk-path) cargo build --profile="${{ env.PROFILE }}" --no-default-features --target=x86_64-apple-darwin

      - name: Create Universal Binary
```

File: .github/workflows/release.yaml [hunk 10]
```diff
@@ -262,103 +257,67 @@ jobs:
...
         run: |
-          mkdir -p "${{ env.OUT_DIR }}"
-          cp "target/${{ env.TARGET }}/${{ env.PROFILE }}/${{ env.BINARY }}" "${{ env.OUT_DIR }}"
-          cp -r assets "${{ env.OUT_DIR }}" || true # Ignore error if `assets` folder does not exist
-
-      - name: Compress package
-        working-directory: ./build/${{ env.PACKAGE }}
+          lipo -create -output target/${{ env.PROFILE }}/${{ env.BINARY }} target/aarch64-apple-darwin/${{ env.PROFILE }}/${{ env.BINARY }} target/x86_64-apple-darwin/${{ env.PROFILE }}/${{ env.BINARY }}
+      - name: Create release
```
Comment (benfrankel) [inline .github/workflows/release.yaml:275]:
```suggestion
          lipo -create -output target/${{ env.PROFILE }}/${{ env.BINARY }} target/aarch64-apple-darwin/${{ env.PROFILE }}/${{ env.BINARY }} target/x86_64-apple-darwin/${{ env.PROFILE }}/${{ env.BINARY }}

      - name: Prepare package
```

File: .github/workflows/release.yaml [hunk 11]
```diff
@@ -262,103 +257,67 @@ jobs:
...
+                    <key>CFBundlePackageType</key>
+                    <string>APPL</string>
+                    <key>CFBundleSupportedPlatforms</key>
+                    <array>
+                        <string>MacOSX</string>
+                    </array>
+                </dict>
+            </plist>
+          EOF
+          cp -r assets/ "build/${{ env.PACKAGE_NAME }}.app/Contents/MacOS/assets"
```
Comment (benfrankel) [inline .github/workflows/release.yaml:308]: I'd add the `|| true # comment` pattern from the other jobs here as well.

File: .github/workflows/release.yaml [hunk 12]
```diff
@@ -262,103 +257,67 @@ jobs:
...
+          cp target/${{ env.PROFILE }}/${{ env.BINARY }} "build/${{ env.PACKAGE_NAME }}.app/Contents/MacOS/"
```
Comment (benfrankel) [inline .github/workflows/release.yaml:309]:
```suggestion
          cp "target/${{ env.PROFILE }}/${{ env.BINARY }}" "build/${{ env.PACKAGE_NAME }}.app/Contents/MacOS/"
```

File: .github/workflows/release.yaml [hunk 13]
```diff
@@ -262,103 +257,67 @@ jobs:
...
+          ln -s /Applications build/
```
Comment (benfrankel) [inline .github/workflows/release.yaml:310]: I'll believe that this is needed, but this looks weird. Any chance you know what this is for? :sweat_smile:

File: .github/workflows/release.yaml [hunk 14]
```diff
@@ -262,103 +257,67 @@ jobs:
...
+          hdiutil create -fs HFS+ -volname "${{ env.PACKAGE_NAME }}" -srcfolder build ${{ env.PACKAGE }}.dmg
```
Comment (benfrankel) [inline .github/workflows/release.yaml:311]: The other jobs would have this as a separate step named `Compress package`. Also I'd rather this output the `.dmg` back into `build/` like the other jobs. Not a big deal though.

File: .github/workflows/release.yaml [hunk 15]
```diff
@@ -262,103 +257,67 @@ jobs:
...
+
+      - name: Upload release
         uses: svenstaro/upload-release-action@v2
         with:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: build/${{ env.PACKAGE }}.dmg
+          file: ${{ env.PACKAGE }}.dmg
```
Comment (benfrankel) [inline .github/workflows/release.yaml:317]:
```suggestion
          file: build/${{ env.PACKAGE }}.dmg
```

**Review by benfrankel — CHANGES_REQUESTED** [review]
Upload package to artifacts is missing for the MacOS job, so its release will not be uploaded to itch.io.

File: .github/workflows/release.yaml (hunk 13 above)
Comment (rparrett) [reply .github/workflows/release.yaml:310]: I think that's originally from my packaging in `pixie_wrangler`. I **think** this is about the "drag and drop the thing into the `Applications` folder" stuff, but I am not an expert here and haven't looked at that code in ages.

Comment (rparrett) [reply .github/workflows/release.yaml:310]: We also probably shouldn't assume that my old stuff is even working.

Comment (janhohenheim) [reply .github/workflows/release.yaml:315]: This is from me! The `hdiutil` assumes it's running from / to search for /Applications for some reason, so this is a little workaround.

File: .github/workflows/release.yaml (hunk 14 above)
Comment (janhohenheim) [reply .github/workflows/release.yaml:316]: Creating a .dmg is not quite the same as compressing. I mean, technically it is, but semantically it creates an actual macOS app.

Comment (benfrankel) [reply .github/workflows/release.yaml:316]: Hm okay, fair enough.

File: .github/workflows/release.yaml [hunk 16]
```diff
@@ -262,103 +257,72 @@ jobs:
...
+                    <array>
+                        <string>MacOSX</string>
+                    </array>
+                </dict>
+            </plist>
+          EOF
+          cp -r assets/ "build/${{ env.PACKAGE_NAME }}.app/Contents/MacOS/assets"
+          cp "target/${{ env.PROFILE }}/${{ env.BINARY }}" "build/${{ env.PACKAGE_NAME }}.app/Contents/MacOS/"
+          ln -s /Applications build/
+          hdiutil create -fs HFS+ -volname "${{ env.PACKAGE_NAME }}" -srcfolder build ${{ env.PACKAGE }}.dmg
```
Comment (benfrankel) [inline .github/workflows/release.yaml:316]:
```suggestion
          hdiutil create -fs HFS+ -volname "${{ env.PACKAGE_NAME }}" -srcfolder build "${{ env.PACKAGE }}.dmg"
```

File: .github/workflows/release.yaml (hunk 13 above)
Comment (benfrankel) [reply .github/workflows/release.yaml:315]: Okay. Maybe in a future PR we can try to find a way to not have to do this, if possible. For now, if it works it works.

File: .github/workflows/release.yaml (hunk 11 above)
Comment (janhohenheim) [reply .github/workflows/release.yaml]: [`1d8f183` (#112)](https://github.com/TheBevyFlock/bevy-template/pull/112/commits/1d8f183e6305acbb4c480f7e62893d3d532f1810)

File: .github/workflows/release.yaml [hunk 17]
```diff
@@ -262,91 +257,69 @@ jobs:
...
-      - name: Compress package
-        working-directory: ./build/${{ env.PACKAGE }}
-        run: |
-          hdiutil create -fs HFS+ -volname "${{ env.PACKAGE_NAME }}" -srcfolder "${{ env.PACKAGE_NAME }}.app" "../${{ env.PACKAGE }}.dmg"
 
       - name: Upload package to artifacts
         uses: actions/upload-artifact@v4
         with:
-          path: build/${{ env.PACKAGE }}.dmg
+          path: "build/${{ env.PACKAGE }}.dmg"
```
Comment (benfrankel) [inline .github/workflows/release.yaml:322]:
```suggestion
          path: build/${{ env.PACKAGE }}.dmg
```
I think quotes don't work here.

**Review by benfrankel — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml [hunk 18]
```diff
@@ -262,86 +257,64 @@ jobs:
...
         run: |
-          hdiutil create -fs HFS+ -volname "${{ env.PACKAGE_NAME }}" -srcfolder "${{ env.PACKAGE_NAME }}.app" "../${{ env.PACKAGE }}.dmg"
-
-      - name: Upload package to artifacts
-        uses: actions/upload-artifact@v4
-        with:
-          path: build/${{ env.PACKAGE }}.dmg
-          name: ${{ env.PLATFORM }}
-          retention-days: 1
+          SDKROOT=$(xcrun -sdk macosx --show-sdk-path) cargo build --profile="${{ env.PROFILE }}" --no-default-features --target=aarch64-apple-darwin
```
Comment (benfrankel) [inline .github/workflows/release.yaml:264]:
```suggestion
          SDKROOT="$(xcrun -sdk macosx --show-sdk-path)" cargo build --profile="${{ env.PROFILE }}" --no-default-features --target=aarch64-apple-darwin
```

**Review by benfrankel — CHANGES_REQUESTED** [review]

**Review by benfrankel — APPROVED** [review]

**Pull request merged**